docs/compare-evidence-strings.md

```diff
@@ -0,0 +1,73 @@
+# Comparison of evidence strings
+
+When major updates to the pipeline are implemented, an important measure of control is comparing evidence strings before and after the update for the same input data. This protocol contains commands which help do this.
+
+## Install JQ locally
+JQ is a command line JSON processor.
+```bash
+wget -q -O jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
+chmod a+x jq
+```
+
+## Sort keys in evidence strings
+This makes them more readable and helps comparison through word diff.
+```bash
+sort_keys () {
+  cat $1 \
+    | ./jq -S "." --tab \
+    | tr -d '\t\n' \
+    | sed -e 's|}{|}~{|g' \
+    | tr '~' '\n' \
+  > $2
+}
+export -f sort_keys
+
+sort_keys /nfs/production3/eva/opentargets/batch-2020-02/evidence_strings/evidence_strings.json old.json &
+sort_keys /nfs/production3/eva/opentargets/batch-2020-02-experimental/evidence_strings/evidence_strings.json new.json
+```
+
+# Extract some fields from the evidence strings
+The fields being extracted are:
+* ClinVar RCV accession
+* Variant ID (rsID or, if absent, RCV accession)
+* Functional consequence SO code
+```
+extract_fields () {
+  cat $1 \
+  | ./jq '.evidence.variant2disease.provenance_type.database.dbxref.url + ">" + .variant.id + "|" + .evidence.gene2variant.functional_consequence' \
+  | tr -d '"' | tr '|' '\t' \
+  | sed -e 's|http://purl.obolibrary.org/obo/||g' \
+        -e 's|http://identifiers.org/clinvar.record/||g' \
+        -e 's|http://identifiers.org/dbsnp/||g' \
+        -e 's|http://www.ncbi.nlm.nih.gov/clinvar/||g' \
+  > $1.fields
+}
+export -f extract_fields
+extract_fields old.json & extract_fields new.json
+```
```

Comment:
Could you try investigating [json-diff](https://pypi.org/project/json-diff/) to check if that makes these (and the steps below) simpler?